assets/model_monitoring/components/src/model_performance_metrics/compute_metrics.py

```diff
@@ -0,0 +1,189 @@
+
+
+class Evaluator:
+    """Evaluator for Compute Metrics mode."""
+
+    def __init__(self, task_type, metrics_config):
+        """
+        Initialize evaluator.
+        Args:
+            task_type: string
+            metrics_config: dict
+        """
+        self.task_type = task_type
+        self.metrics_config = metrics_config
+
+    @abstractmethod
+    def evaluate(self, metrics_dto, **kwargs):
+        """Evaluate predictions.
+
+        Args:
+            metrics_dto: metrics dto
+
+        Returns:
+            metrics: dict
+        """
+        pass
+
+    def _convert_predictions(self, preds):
+        """Convert predictions to np array.
+
+        Args:
+            preds: predictions
+
+        Returns: np array of predictions
+
+        """
+        if isinstance(preds, pd.DataFrame) and len(preds.columns) == 1:
+            return preds[preds.columns[0]].to_numpy()
+        if isinstance(preds, pd.DataFrame) or isinstance(preds, pd.Series):
+            return preds.to_numpy()
+        if isinstance(preds, list):
+            return np.array(preds)
+        return preds
+
+
+class ClassifierEvaluator(Evaluator):
+
+    def __init__(self, task_type, metrics_config):
+        """
+        Initialize classifier evaluator.
+        Args:
+            task_type: string
+            metrics_config: dict
+        """
+        super().__init__(task_type, metrics_config)
+
+    def evaluate(self, metrics_dto, **kwargs):
+        """Evaluate classification.
+
+        Args:
+            metrics_dto: metrics dto
+
+        Returns: metrics
+        """
+        y_pred = self._convert_predictions(metrics_dto.predictions)
+        y_test = self._convert_predictions(metrics_dto.ground_truth)
+        if "metrics" not in self.metrics_config:
+            self.metrics_config["metrics"] = [Metric.Accuracy, Metric.PrecisionMacro, Metric.RecallMacro,
+                                              Metric.F1Macro]
+        metrics = compute_metrics(task_type=constants.Tasks.CLASSIFICATION, y_test=y_test, y_pred=y_pred
+                                  , **self.metrics_config)
+        return metrics
+
+
+class RegressorEvaluator(Evaluator):
+    """Regressor Evaluator.
+
+    Args:
+        Evaluator (_type_): _description_
+    """
+
+    def __init__(self, task_type, metrics_config):
+        """__init__.
+
+        Args:
+            task_type: string
+            metrics_config: dict
+        """
+        super().__init__(task_type, metrics_config)
+
+    def evaluate(self, metrics_dto, **kwargs):
+        """
+
+        Args:
+            metrics_dto: metrics dto
+            **kwargs:
+
+        Returns: metrics
+
+        """
+        y_pred = self._convert_predictions(metrics_dto.predictions)
+        y_test = self._convert_predictions(metrics_dto.ground_truth)
+        if "metrics" not in self.metrics_config:
+            self.metrics_config["metrics"] = [Metric.RMSE, Metric.MeanAbsError]  # Metric.MSE
```

Comment:
Why do not we have Metric.MSE? I saw it was commented off

Reply:
I think this is not required for now. Let me know if we need this metric as well